csv_export_partner/models/csv_export_partner.py

```diff
@@ -55,24 +55,40 @@ class PartnerCSVExport(models.TransientModel):
     _filename_template = "CLI_%Y%m%d_%H%M_%S%f.csv"
 
     def get_recordset(self):
-        # cf csv_export_invoice.py
-        exported_invoices = self.env["account.invoice"].search(
-            [
-                ("state", "!=", "draft"),
-                ("state", "!=", "cancel"),
-                ("date", ">=", self.start_date),
-                ("date", "<", self.end_date),
-            ]
-        )
-        # cf csv_export_payment.py
-        exported_payments = self.env["account.payment"].search(
-            [
-                ("journal_id.type", "=", "cash"),
-                ("state", "!=", "draft"),
-                ("payment_date", ">=", self.start_date),
-                ("payment_date", "<", self.end_date),
-            ]
-        )
+        if self.manual_date_selection:
```

Comment:
same style comment here that will allow you to write :

```py
invoice_domain = [("state", "!=", "draft"), ("state", "!=", "cancel")]
payment_domain = [("journal_id.type", "=", "cash"), ("state", "!=", "draft")]
if self.manual_date_selection:
     invoice_domain += [("date", ">=", self.start_date), ("date", "<", self.end_date)]
    payment_domain += [("payment_date", ">=", self.start_date), ("payment_date", "<", self.end_date)]
else:
    exported_domain = ("export_to_sftp", "=", False)
    invoice_domain.append(exported_domain)
    payment_domain.append(exported_domain)

exported_invoices = self.env["account.invoice"].search(invoice_domain)
exported_payments = self.env["account.payment"].search(payment_domain)
```

Reply:
fixed